### PR TITLE
Update subscriptions-core to `1.4.0`.

### PR DESCRIPTION
**Pull request by shendy-a8c**
#### Changes proposed in this Pull Request

- Load newer woocommerce-subscriptions-core's version, `1.4.0`.
- Add changes of woocommerce-subscriptions-core to the changelog.
- Updated `composer.lock` as a result of running `composer update woocommerce/subscriptions-core`.

#### Testing instructions

* Checkout this branch.
* [Husky hook should run `composer install`](https://github.com/Automattic/woocommerce-payments/blob/develop/.husky/post-checkout#L8) and your `vendor/woocommerce/subscriptions-core` should be updated to be `1.4.0`. Confirm that [latest changes from `1.4.0`](https://github.com/Automattic/woocommerce-subscriptions-core/commits/1.4.0) is present, for example, [vendor/woocommerce/subscriptions-core/changelog.txt is up-to-date](https://github.com/Automattic/woocommerce-subscriptions-core/blob/1.4.0/changelog.txt).

-------------------

- [x] Added changelog entries to both `readme.txt` and `changelog.txt` (or does not apply)
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Added testing instructions to the [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) (or does not apply)

Closes https://github.com/Automattic/woocommerce-payments/issues/3537.
Closes https://github.com/Automattic/woocommerce-payments/issues/3596.

### CODE REVIEW

**Comment by shendy-a8c** [comment]
Do we want to add testing instructions for GlobalStep for https://github.com/Automattic/woocommerce-payments/issues/3537 and / or https://github.com/Automattic/woocommerce-payments/issues/3596?

File: composer.lock [hunk 1]
```diff
@@ -4,25 +4,25 @@
         "Read more about it at https://getcomposer.org/doc/01-basic-usage.md#installing-dependencies",
         "This file is @generated automatically"
     ],
-    "content-hash": "5fd224bcb4d9bea59b2b592336ca670b",
+    "content-hash": "fd396360a87fc51fe044306a6490d821",
     "packages": [
         {
             "name": "automattic/jetpack-a8c-mc-stats",
-            "version": "v1.4.10",
+            "version": "v1.4.11",
```
Comment (brucealdridge) [inline composer.lock:11]: looks like a few other packages got updated as well.

I'm assuming this was unintentional.

**Review by brucealdridge — COMMENTED** [review]
Looks like ~10 packages were updated rather than just subscriptions-core

**Comment by haszari** [comment]
> Do we want to add testing instructions for GlobalStep for #3537 and / or #3596?

Sounds like a good idea since these are both somewhat serious regressions. CC @mattallan (WCPay 3.6 Helix DRI)

**Comment by shendy-a8c** [comment]
> Looks like ~10 packages were updated rather than just subscriptions-core

Thank you @brucealdridge for catching that. I have been running `composer update` which doesn't only update `woocommerce/subscriptions-core`.
I should have run `composer update woocommerce/subscriptions-core`, right?
I'll fix after we re-tag subscriptions-core 1.4.0.

**Comment by brucealdridge** [comment]
> Thank you @brucealdridge for catching that. I have been running `composer update` which doesn't only update `woocommerce/subscriptions-core`.
> I should have run `composer update woocommerce/subscriptions-core`, right?

Correct. `composer update` by itself is somewhat dangerous. It always pays to double check the output of the command and make sure it's only updating what you expect.

**Comment by shendy-a8c** [comment]
Done:
- Merge `develop` to this PR branch and resolve conflicts.
- Updated `composer.lock` by running `composer update woocommerce/subscriptions-core` instead of `composer update`.

Ready for another round of review.

**Pull request merged**